core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1651,7 +1651,7 @@ private[spark] object Utils extends Logging {
   /**
    * Format a Windows path such that it can be safely passed to a URI.
    */
-  def formatWindowsPath(path: String): String = path.replace("\\", "/")
+  def formatWindowsPath(path: String): String = path.replace("\\", "/").replace(" ", "%20")
```

Comment:
Hm, is this specific to Windows, even? if I had a path with a space on Linux it would fail similarly, I'd imagine? I bet we dont' get that case right either.